src/perovskite_tandem_database/parsers/tandemparser.py

```diff
@@ -0,0 +1,867 @@
+import re
+from typing import TYPE_CHECKING
+from unicodedata import numeric
+
+import pandas as pd
+from nomad.config import config
+from nomad.datamodel.data import EntryData
+
+# from nomad.parsing.tabular import create_archive
+from nomad.datamodel.datamodel import EntryArchive
+from nomad.datamodel.metainfo.annotations import ELNAnnotation
+from nomad.metainfo import Quantity
+from nomad.parsing.parser import MatchingParser
+from pint import UnitRegistry, errors
+
+from perovskite_tandem_database.parsers.utils import create_archive
+from perovskite_tandem_database.schema_packages.schema import PerovskiteTandemSolarCell
+from perovskite_tandem_database.schema_packages.tandem import (
+    ChalcopyriteAlkaliMetalDoping,
+    ChalcopyriteLayer,
+    Cleaning,
+    CleaningStep,
+    GasPhaseSynthesis,
+    General,
+    Ion,
+    Layer,
+    LiquidSynthesis,
+    NonAbsorbingLayer,
+    PerovskiteComposition,
+    PerovskiteLayer,
+    PhotoAbsorber,
+    QuenchingSolvent,
+    ReactionComponent,
+    Reference,
+    SiliconLayer,
+    Solvent,
+    Storage,
+    SubCell,
+    Substance,
+    Substrate,
+    Synthesis,
+    SynthesisStep,
+    ThermalAnnealing,
+)
+
+if TYPE_CHECKING:
+    from nomad.datamodel.datamodel import EntryArchive
+    from structlog.stdlib import BoundLogger
+
+
+class TandemParser(MatchingParser):
+    """
+    Parser for matching tandem db files and creating instances of PerovskiteTandemSolarCell.
+    """
+
+    def parse(
+        self,
+        mainfile: str,
+        archive: 'EntryArchive',
+        logger: 'BoundLogger',
+        child_archives: dict[str, 'EntryArchive'] = None,
+    ) -> None:
+        logger.info('TandemParser.parse')
+        data_frame = pd.read_excel(mainfile, index_col=0)
+
+        # Process each column/device/publication separately
+        for col in data_frame.columns:
+            logger.info(f'Processing column: {col}')
+
+            # Clean the data frame
+            # Set proper Boolean values and remove rows with all NaN values
+            column_data = cleanup_dataframe(data_frame[col])
+
+            stack = extract_layer_stack(column_data)
+            general = extract_general(column_data)
+            reference = extract_reference(column_data)
+            reference.normalize(archive, logger)
```

Comment:
Remove this line. This happens automatically afterwards. `   reference.normalize(archive, logger)`